packages/go_router/lib/src/route.dart

```diff
@@ -227,3 +145,55 @@ class GoRoute {
   ) =>
       const SizedBox.shrink();
 }
+
+/// A route that displays a UI shell around the matching child route.
+///
+/// The widget built by the matching child route becomes to the child parameter
+/// of the [builder].
+class ShellRoute extends RouteBase {
+  /// Constructs a [ShellRoute].
+  ShellRoute({
+    required String path,
+    required this.builder,
+    this.defaultRoute,
+    GoRouterRedirect redirect = RouteBase._emptyRedirect,
+    List<RouteBase> routes = const <RouteBase>[],
+  }) : super._(
+          path: path,
+          routes: routes,
+          redirect: redirect,
+        );
+
+  /// The widget builder for a shell route.
+  final ShellRouteBuilder builder;
+
+  /// The relative path to the child route to navigate to when this route is
+  /// displayed. This allows the default child route to be specified without
+  /// using redirection.
+  final String? defaultRoute;
+}
+
+/// A route that displays all descendent [StackedRoute]s within its visual
+/// boundary, typically the UI shell of a [ShellRoute].
+///
+/// This route places a nested [Navigator] in the widget tree, where any
```

Comment:
A navigator or a router? the difference is how we are going to handle backbutton

Reply:
Can you explain more what you mean?